compiler/src/main/java/org/neo4j/ogm/compiler/MultiStatementCypherCompiler.java

```diff
@@ -100,10 +102,8 @@ public void unmap(NodeBuilder nodeBuilder) {
 		List<Statement> statements = new ArrayList<>(newNodesByLabels.size());
 		for (Set<Node> nodeModels : newNodesByLabels.values()) {
 			NewNodeEmitter newNodeEmitter = new NewNodeEmitter(nodeModels);
-			final Map<String, Object> parameters = new HashMap<>();
-			final StringBuilder query = new StringBuilder();
-			newNodeEmitter.emit(query, parameters);
-			statements.add(statementFactory.statement(query.toString(), parameters));
+			Pair<String, Map<String, Object>> emittedCypher =  newNodeEmitter.emit();
+			statements.add(statementFactory.statement(emittedCypher.first, emittedCypher.second));
```

Comment:
Yeah.. The API wasn't right to begin with and I think at the moment I've left it in a half done state. Shouldn't take too long to just do it right.